src/main/java/com/google/api/gax/bundling/ThresholdBundler.java

```diff
@@ -47,34 +46,36 @@
  * Queues up elements until either a duration of time has passed or any threshold in a given set of
  * thresholds is breached, and then delivers the elements in a bundle to the consumer.
  */
-public final class ThresholdBundler<E> {
+public final class ThresholdBundler<E extends Bundle<E>> {
 
   private ImmutableList<BundlingThreshold<E>> thresholdPrototypes;
   private final Duration maxDelay;
   private final BundlingFlowController<E> flowController;
+  private final BundleFactory<E> bundleFactory;
 
   private final Lock lock = new ReentrantLock();
   private final Condition bundleCondition = lock.newCondition();
-  private Bundle currentOpenBundle;
-  private List<Bundle> closedBundles = new ArrayList<>();
+  private TrackedBundle currentOpenTrackedBundle;
```

Comment:
This PR is a little difficult to review because of renames. Nothing to do right now. In the future, I think we should consider moving big renames into its own PR. This is definitely not just you; I'm guilty of this too.